examples/AN00221_freq_domain_example/src/AN00218_freq_domain_example.xc

```diff
@@ -60,23 +134,36 @@ void freq_domain_example(streaming chanend c_ds_output[2]){
 
         while(1){
 
-           //Recieve the preprocessed frames ready for the FFT
+            lib_dsp_fft_complex_t p[FFT_N]; // use as tmp buffer as well as output buffer
+
+#if INPUT_FROM_HW
+            //Recieve the preprocessed frames ready for the FFT
             mic_array_frame_fft_preprocessed * current = mic_array_get_next_frequency_domain_frame(c_ds_output, 2, buffer, comp, dc);
 
+#if DSP_ENABLED
            for(unsigned i=0;i<MIC_ARRAY_NUM_FREQ_CHANNELS;i++){
+#if MIC_ARRAY_WORD_LENGTH_SHORT
+               // convert current->data[i] into p
```

Comment:
This if def stuff is very confusing for an app note. I would suggest removing INPUT_FROM_HW and DSP_ENABLED as the demo requires both